### ISSUE TITLE
SPIKE: Does it make sense to use aria-describedby for Details links

### ISSUE THREAD

**Issue by benbrasso-agile6** [department-of-veterans-affairs/va.gov-team#94424]
## Description 

Use `aria-describedby` on the details links instead of `aria-label`. Feedback from 508 Office ([Slack thread](https://dsva.slack.com/archives/C03CGTDLTFF/p1727984103633899).)

`Details for [appt type] [care type] appointment on [Friday, October 4 10:55 a.m., Mountain time]`

^ Note: VistA appts from CHIP don't have timezone 

## Tasks

- [ ] Use `aria-describedby` on any details links instead of `aria-label`. Includes eCheck-in and pre-check-in. Appts on upcoming appt list page. And, details link in the task card on the appt list page.

**Comment by benbrasso-agile6** [comment]
@loripusey FYI

**Comment by loripusey** [comment]
added to next sprint

**Comment by brianseek** [comment]
This should get broken up into a ticket for CIE and one for VAOS

**Comment by benbrasso-agile6** [comment]
@outerpress and @ldelacosta are taking care of Appts application

**Comment by benbrasso-agile6** [comment]
Added `upcoming` to better clarify for PCI apps

**Comment by wullaski** [comment]
@loripusey I had some time and dug into this a bit. We can use the aria-describedby but readout will be a bit more stilted. We won't be able to have "Details for Mental health appointment on Fri, February 28th at 12:30 p.m." it will be more like "Details, Mental health with Dr. Jones, 28, Fri, 12:30 p.m."

**Comment by loripusey** [comment]
@department-of-veterans-affairs/uae-ux any thoughts on this change?

**Comment by outerpress** [comment]
Summoning @rileyorr 's a11y powers on this one. Riley, I think we're okay to lose the word appointment since you have that context from the page, but the date seems a little clunky/hard to parse. Thoughts?
